go/adbc/driver/bigquery/driver.go

```diff
@@ -0,0 +1,222 @@
+
+import (
+	"cloud.google.com/go/bigquery"
+	"fmt"
+	"github.com/apache/arrow-adbc/go/adbc"
+	"github.com/apache/arrow-adbc/go/adbc/driver/internal/driverbase"
+	"github.com/apache/arrow/go/v16/arrow/memory"
+	"golang.org/x/exp/maps"
+	"runtime/debug"
+	"strings"
+)
+
+// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+const (
+	OptionStringAuthType               = "adbc.bigquery.sql.auth_type"
+	OptionStringCredentials            = "adbc.bigquery.sql.credentials"
+	OptionStringProjectID              = "adbc.bigquery.sql.project_id"
+	OptionValueAuthTypeDefault         = "default"
+	OptionValueAuthTypeCredentialsFile = "credentials_file"
+
+	OptionStringQueryDestinationTable  = "adbc.bigquery.sql.query.destination_table"
+	OptionStringQueryDefaultProjectID  = "adbc.bigquery.sql.query.default_project_id"
+	OptionStringQueryDefaultDatasetID  = "adbc.bigquery.sql.query.default_dataset_id"
+	OptionStringQueryCreateDisposition = "adbc.bigquery.sql.query.create_disposition"
+	OptionStringQueryWriteDisposition  = "adbc.bigquery.sql.query.write_disposition"
+	OptionBoolQueryDisableQueryCache   = "adbc.bigquery.sql.query.disable_query_cache"
+	OptionBoolDisableFlattenedResults  = "adbc.bigquery.sql.query.disable_flattened_results"
+	OptionBoolQueryAllowLargeResults   = "adbc.bigquery.sql.query.allow_large_results"
+	OptionStringQueryPriority          = "adbc.bigquery.sql.query.priority"
+	OptionIntQueryMaxBillingTier       = "adbc.bigquery.sql.query.max_billing_tier"
+	OptionIntQueryMaxBytesBilled       = "adbc.bigquery.sql.query.max_bytes_billed"
+	OptionBoolQueryUseLegacySQL        = "adbc.bigquery.sql.query.use_legacy_sql"
+	OptionBoolQueryDryRun              = "adbc.bigquery.sql.query.dry_run"
+	OptionBoolQueryCreateSession       = "adbc.bigquery.sql.query.create_session"
+	OptionIntQueryJobTimeout           = "adbc.bigquery.sql.query.job_timeout"
+)
+
+var (
+	infoVendorVersion string
+)
+
+func init() {
+	if info, ok := debug.ReadBuildInfo(); ok {
+		for _, dep := range info.Deps {
+			switch {
+			case dep.Path == "cloud.google.com/go/bigquery":
+				infoVendorVersion = dep.Version
+			}
+		}
+	}
+}
+
+type driverImpl struct {
+	driverbase.DriverImplBase
+	alloc memory.Allocator
```

Comment:
same as before, `Alloc` already exists in `DriverImplBase` so you don't need another one here